src/typing/type_inference_js.ml

```diff
@@ -876,7 +876,8 @@ let rec convert cx map = Ast.Type.(function
     (* Do not evaluate existential type variables when map is non-empty. This
        ensures that existential type variables under a polymorphic type remain
        unevaluated until the polymorphic type is applied. *)
-    let force = SMap.is_empty map in
+    (* SG: Does this logic still make sense with scoped type variables? *)
+    let force = SMap.is_empty type_params_map in
```

Comment:
Not sure if this check will still serve the intended purpose with scoped type vars.